tests/test_outlines.py

```diff
@@ -0,0 +1,264 @@
+from itertools import repeat
+
+import pytest
+from hypothesis import given, example
+from hypothesis import strategies as st
```

Comment:
I'm super impressed when anyone submit a PR with tests. I usually have to write them.... Thanks.

Reply:
Thank you for your feedback!

Using the existing libqpdf helpers would most certainly be a good idea. Safety measures against circular references could be implemented in Python, but if they are already part of libqpdf it might be better to just use these.
Currently I have also not looked into resolving named destinations, since I first wanted to get the most-simple use case right, i.e. creating an outline from scratch and referring to pages. For copying outlines from one document to another this would be required however; using any present libqpdf functionality should help here as well.

It should be possible to combine the implementations somehow. Since I am not experienced in interfacing C++ code I can however not provide much support for creating the bindings.

Reply:
No problem.